### PR TITLE
add assetsearch and pull in assetswithcode

### PR DESCRIPTION
**Pull request by xgess**
added a new `AssetSearch` method that takes an arg object. i figured this would be easier to expand on than having to change the method signature. This method does not error if it cannot find any matches. 
also pulled in `AssetsWithCode` for a little more code sharing. Left the behavior of erroring when it can't find a match.

### CODE REVIEW

File: assets_test.go [hunk 1]
```diff
@@ -30,3 +30,101 @@ func TestAsset(t *testing.T) {
...
+		NumAccounts:             3,
+	}
+	var foundMatch bool
+	for _, asset := range res {
+		if asset == expectedMatch {
+			foundMatch = true
+			break
+		}
+	}
+	require.True(t, foundMatch)
```
Comment (zapu) [inline assets_test.go]: would `require.Contains` work here?

Comment (xgess) [reply assets_test.go]: 🎉 yes it will! update coming.

File: assets_test.go [hunk 2]
```diff
@@ -30,3 +30,101 @@ func TestAsset(t *testing.T) {
...
+	foundMatch = false
+	for _, asset := range res {
+		if asset == expectedMatch {
+			foundMatch = true
+			break
+		}
+	}
+	require.True(t, foundMatch)
+	require.Equal(t, len(res), 1)
+
```
Comment (zapu) [inline assets_test.go]: can you add a test for credit_alphanum12 asset as well?

Comment (xgess) [reply assets_test.go]: done

**Comment by xgess** [comment]
@zapu agrees. this passes locally.

**Pull request merged**